app/views/directives/osc-webhook-triggers.html

```diff
@@ -0,0 +1,82 @@
+<ng-form name="secretsForm" class="add-webhook">
+  <div ng-repeat="trigger in webhookTriggers">
+    <div class="add-webhook-row">
+      <ui-select 
+        ng-model="trigger.data.type"
+        ng-disabled="isDeprecated(trigger)"
+        on-select="triggerTypeChange(trigger)"
+        on-remove="test()"
+        search-enabled="false"
+        title="Select a webhooke type"
+        class="select-webhook-type"
+        flex>
+        <ui-select-match placeholder="Webhook type">
+          {{ $select.selected.label }}
+        </ui-select-match>
+        <ui-select-choices repeat="option.label as option in webhookTypesOptions">
+          {{ option.label }}
+        </ui-select-choices>
+      </ui-select>
+
+      <ui-select
+        ng-if="!isDeprecated(trigger)"
+        ng-model="trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
+        ng-disabled="!trigger.data.type"
+        ng-required="trigger.data.type"
+        on-select="triggerSecretChange(trigger)"
+        title="Select a webhooke secret reference"
+        class="select-secret-ref"
+        flex>
+        <ui-select-match placeholder="Webhook secret reference">
+          {{ $select.selected.metadata.name }}
+        </ui-select-match>
+        <ui-select-choices repeat="webhookSecret.metadata.name as webhookSecret in (webhookSecrets | filter : $select.search)">
+          <div ng-bind-html="webhookSecret.metadata.name | highlight : $select.search"></div>
+        </ui-select-choices>
+      </ui-select>
+
+      <div class="select-secret-ref deprecated-secret input-group" ng-if="isDeprecated(trigger)">
+        <input
+          ng-model="trigger.data[trigger.data.type.toLowerCase()].secret"
+          class="form-control"
+          type="password"
+          autocorrect="off"
+          autocapitalize="none"
+          spellcheck="false"
+          select-on-focus
+          disabled>
+        <span class="input-group-btn">
+          <a
```

Comment:
button instead of a?

Reply:
button would submit the form in which we put the directive/component
we are using the anchor tag in the [copy-to-clipboard](https://github.com/openshift/origin-web-console/blob/master/app/views/directives/_copy-to-clipboard.html#L14) directive as well

Reply:
We're trying to move to buttons when it makes sense because they're better for accessibility.

Does adding `type="button"` fix the form submit problem?

Reply:
yes, updated !